node/src/components/consensus/protocols/highway.rs

```diff
@@ -774,7 +710,9 @@ where
     }
 
     fn has_received_messages(&self) -> bool {
-        !self.highway.state().is_empty() || !self.vertex_deps.is_empty()
+        !self.highway.state().is_empty()
+            || !self.synchronizer.is_empty()
+            || !self.pending_values.is_empty()
```

Comment:
In which scenario we would have `pending_values` but nothing in the `synchronizer`?

Reply:
I kept `pending_values` in `HighwayProtocol` itself. So:
If someone sends us a vertex with a block, it first goes into the synchronizer, until all dependencies are satisfied. Then it goes out of the synchronizer and into `pending_vertices` until the block validator responds.